src/Nethermind/Nethermind.JsonRpc/Data/ForkActivationParameter.cs

```diff
@@ -0,0 +1,190 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using Nethermind.Core.Specs;
+using Nethermind.Specs.Forks;
+
+namespace Nethermind.JsonRpc.Data;
+
+/// <summary>
+/// Parameter for specifying fork specification overrides for trace operations.
+/// Allows tracing blocks with different protocol rules than their original fork.
+/// 
+/// Examples:
+/// - {"forkName": "osaka"} - Use Osaka fork rules
+/// - {"activationBlock": 1000000} - Override activation block
+/// - {"forkName": "prague", "activationBlock": 0} - Use Prague from genesis
+/// </summary>
+public class ForkActivationParameter
+{
+    private static readonly Dictionary<string, Func<IReleaseSpec>> ForkRegistry = new(StringComparer.OrdinalIgnoreCase)
+    {
+        ["frontier"] = () => Frontier.Instance,
+        ["homestead"] = () => Homestead.Instance,
+        ["tangerinewhistle"] = () => TangerineWhistle.Instance,
+        ["spuriousdragon"] = () => SpuriousDragon.Instance,
+        ["byzantium"] = () => Byzantium.Instance,
+        ["constantinople"] = () => Constantinople.Instance,
+        ["constantinoplefix"] = () => ConstantinopleFix.Instance,
+        ["istanbul"] = () => Istanbul.Instance,
+        ["muirglacier"] = () => MuirGlacier.Instance,
+        ["berlin"] = () => Berlin.Instance,
+        ["london"] = () => London.Instance,
+        ["arrowglacier"] = () => ArrowGlacier.Instance,
+        ["grayglacier"] = () => GrayGlacier.Instance,
+        ["paris"] = () => Paris.Instance,
+        ["shanghai"] = () => Shanghai.Instance,
+        ["cancun"] = () => Cancun.Instance,
+        ["prague"] = () => Prague.Instance,
+        ["osaka"] = () => Osaka.Instance
+    };
```

Comment:
This could be moved to `MainnetSpecProvider` and potentially differ per `ISpecProvider` implementation as AuRa/Gnosis will have slightly different release specs

Optimism could have widely different ones